### PR TITLE
IP Rotation Extension

### PR DESCRIPTION
**Pull request by ByteAlex**

### CODE REVIEW

File: ROUTEPLANNERS.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# RoutePlanner Strategies
+
+## Terminology
+### Ip Block / Subnet
+An IP Block or Subnet are addresses which are next to each other
+
+(p.e.) 1.0.0.0, 1.0.0.1, 1.0.0.2, ...
```
Comment (freyacodes) [inline ROUTEPLANNERS.md]: What does p.e. mean? Did you mean e.g.?

File: ROUTEPLANNERS.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# RoutePlanner Strategies
+
+## Terminology
+### Ip Block / Subnet
```
Comment (freyacodes) [inline ROUTEPLANNERS.md]: They're not subnets. Call them CIDR blocks instead. Not all CIDR blocks are subnets. For instance, my home router's LAN subnet is `192.168.0.100/24`.

Refer to proper CIDR documentation.

File: ROUTEPLANNERS.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+An IP Block or Subnet are addresses which are next to each other
+
+(p.e.) 1.0.0.0, 1.0.0.1, 1.0.0.2, ...
+
+### Combined IP Block
+A combined IP Block is created when multiple IPs / subnets are added to the 
+configuration file. Lavaplayer will convert them to a single virtual IP-Block.
```
Comment (freyacodes) [inline ROUTEPLANNERS.md]: > convert them to

Better wording might be “treat them like”

File: ROUTEPLANNERS.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+## Strategies
```
Comment (freyacodes) [inline ROUTEPLANNERS.md]: This section is overly terse. Consider removing the 3 H4s and just write a paragraph.

File: ROUTEPLANNERS.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+### RotateOnBan Strategy
+#### RoutePlanner
+This strategy uses the `RotatingIpRoutePlanner`.
+#### Implementation
+This strategy switches the ip as soon as the used address got banned.
+#### Requirements / Recommendations
+The usage of this strategy is recommended for IPv4 blocks or smaller IPv6 blocks 
+(less than a /64)
+
+---
```
Comment (freyacodes) [inline ROUTEPLANNERS.md]: These separators are unnecessary

File: ROUTEPLANNERS.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+as offset in the used block.
+
+When a ban occurs, this strategy rotates to the next /64 block as fallback strategy.
+#### Requirements / Recommendations
+This strategy requires at least a /64 IPv6 subnet (18 quintillion 
+addresses).
+
+For a working rotation you need at least 2 /64 IPv6 subnets.
+
+I don't recommend using combined /64s here (p.e. 2 different /65s). You can
```
Comment (freyacodes) [inline ROUTEPLANNERS.md]: Avoid using the pronoun `I`. The meaning is misleading when I own this repository.

**Review by freyacodes — CHANGES_REQUESTED** [review]
This is just a review of the docs

File: ROUTEPLANNERS.md (hunk 2 above)
Comment (freyacodes) [reply ROUTEPLANNERS.md]: New version does still not explain CIDR notation.

Link to a proper resource.

File: ROUTEPLANNERS.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+as offset in the used block.
+
+When a ban occurs, this strategy rotates to the next /64 block as fallback strategy.
+### Requirements / Recommendations
+This strategy requires at least a /64 IPv6 CIDR (18 quintillion 
+addresses).
+
+For a working rotation you need at least 2 /64 IPv6 CIDRs.
+
+Its not recommended using combined /64s here (e.g. 2 different /65s). But you can
```
Comment (freyacodes) [inline ROUTEPLANNERS.md]:
```suggestion
It is not recommended using combined /64s here (e.g. 2 different /65s), though you can 
```

File: ROUTEPLANNERS.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+### RoutePlanner
+This strategy uses the `NanoIpRoutePlanner`.
+### Implementation
+This strategy switches the ip on each clock-update and uses the current nano-time
+as offset in the used block.
+### Requirements / Recommendations
+This strategy requires at least a (combined) /64 IPv6 CIDR (18 quintillion 
+addresses).
+
+If you have a CIDR which is (combined) bigger than a /64, its recommended to use
```
Comment (freyacodes) [inline ROUTEPLANNERS.md]:
```suggestion
If you have a CIDR which is (combined) bigger than a /64, it is recommended to use
```

File: ROUTEPLANNERS.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+strategy.
+
+This strategy does not require a big CIDR to work, but you will most likely 
+get banned when you have not enough IP's in the used CIDR.
+
+## NanoSwitch Strategy
+### RoutePlanner
+This strategy uses the `NanoIpRoutePlanner`.
+### Implementation
+This strategy switches the ip on each clock-update and uses the current nano-time
```
Comment (freyacodes) [inline ROUTEPLANNERS.md]:
```suggestion
This strategy switches the IP on each clock update and uses the current nanosecond
```

File: ROUTEPLANNERS.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
+# RoutePlanner Strategies
+
+> ***Terminology***
+>
+> **Ip block / CIDR block**
+>
+> An IP block or CIDR block are addresses which are next to each other
+>
+> (e.g.) 1.0.0.0, 1.0.0.1, 1.0.0.2, ...
```
Comment (freyacodes) [inline ROUTEPLANNERS.md]: These are not CIDRs. They're just IPs.

Comment (ByteAlex) [reply ROUTEPLANNERS.md]: are addresses which are next to each other... e.g. address1, address2 and so on. so I wanted to show what I mean by "addresses that are next to each other"

**Review by freyacodes — APPROVED** [review]

**Pull request merged**